MaxText/layers/linears.py

```diff
@@ -643,12 +707,8 @@ def __call__(self, inputs):
         name="gate",
         matmul_precision=self.config.matmul_precision,
     )(inputs)
-
-    w0_kernel, w1_kernel, wo_kernel = self.generate_kernels(cfg.num_experts, cfg.emb_dim, cfg.mlp_dim)
-
+    max_logging.log("Running MoE megablox implementation.")
```

Comment:
Thank you for the change! Hmm.. but the logic seems not right now.

Could we revert it back and change it to something below?

```
    if cfg.megablox:
      max_logging.log("Running MoE megablox implementation.")
      A helper function to do whatever we like to process quantized weights. 
      return self.megablox(inputs, gate_logits, w0_kernel, w1_kernel, wo_kernel)
    else:
      max_logging.log("Running MoE matmul implementation.")
      w0_kernel, w1_kernel, wo_kernel = self.generate_kernels(cfg.num_experts, cfg.emb_dim, cfg.mlp_dim)
      return self.dense_matmul(inputs, gate_logits, w0_kernel, w1_kernel, wo_kernel)
```
I like your previous change, but just wrap those lines into a helper function. Happy to discuss offline :)

Reply:
revert it back to the previous code, and added helper function.

Reply:
Thanks! After moving the logs, it WILL LGTM!